metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -51,39 +54,38 @@ class ScalametaLanguageServer(
 )(implicit s: Scheduler)
     extends LanguageServer(lspIn, lspOut) {
   implicit val cwd: AbsolutePath = config.cwd
-  val (semanticdbSubscriber, semanticdbPublisher) =
+  val (fileSystemSemanticdbSubscriber, fileSystemSemanticdbsPublisher) =
     ScalametaLanguageServer.semanticdbStream(cwd)
   val (compilerConfigSubscriber, compilerConfigPublisher) =
-    Observable.multicast[AbsolutePath](MulticastStrategy.Publish)
-  def onError(e: Throwable): Unit = {
-    logger.error(e.getMessage, e)
-  }
+    ScalametaLanguageServer.compilerConfigStream(cwd)
   val buffers: Buffers = Buffers()
-  val scalac = new ScalacProvider(config, compilerConfigPublisher)
-  val databasePublisher: Observable[Database] = Observable.merge(
-    semanticdbPublisher.doOnError(onError),
-    scalac.documentPublisher.map(doc => Database(doc :: Nil))
-  )
-  val symbolIndexer: SymbolIndex = SymbolIndex(
-    cwd,
-    connection,
-    buffers
-  )
-  val onIndexDatabase: Observable[Effects.IndexSemanticdb] =
-    databasePublisher.map { db =>
-      symbolIndexer.indexDatabase(db)
-      Effects.IndexSemanticdb
-    }
-  val scalafix: Linter = new Linter(
-    cwd,
-    stdout,
-    connection,
-    semanticdbPublisher.map(_.toDb(None)).doOnError(onError)
-  )
+  val scalac: ScalacProvider = new ScalacProvider(config)
+  val symbolIndexer: SymbolIndex = SymbolIndex(cwd, connection, buffers, config)
+  val scalafix: Linter = new Linter(cwd, stdout, connection)
+  val metaSemanticdbs: Observable[semanticdb.Database] =
+    fileSystemSemanticdbsPublisher.map(_.toDb(None))
```

Comment:
_.toDb(sourcePath = None) or just add a default of None and call _.toDb?
_.toDb(None) has that same confusing quality as boolean parameters do.